shotover-proxy/tests/opensearch_int_tests/mod.rs

```diff
@@ -1,89 +1,240 @@
+    let hits = results["hits"]["hits"].as_array().unwrap();
+    assert_eq!(
+        hits.iter().map(|x| &x["_source"]).collect::<Vec<_>>(),
+        vec!(
+            &json!({ "title": "OpenSearch", "i": 0 }),
+            &json!({ "title": "OpenSearch", "i": 1 }),
+            &json!({ "title": "OpenSearch", "i": 2 }),
+            &json!({ "title": "OpenSearch", "i": 3 }),
+            &json!({ "title": "OpenSearch", "i": 4 }),
+            &json!({ "title": "OpenSearch", "i": 5 }),
+            &json!({ "title": "OpenSearch", "i": 6 }),
+            &json!({ "title": "OpenSearch", "i": 7 }),
+            &json!({ "title": "OpenSearch", "i": 8 }),
+            &json!({ "title": "OpenSearch", "i": 9 }),
+        )
+    );
+}
+
+async fn test_create_index(client: &OpenSearch) {
     client
-        .bulk(BulkParts::Index(index))
-        .body(body)
-        .refresh(Refresh::WaitFor)
+        .indices()
+        .create(IndicesCreateParts::Index("test-index"))
+        .send()
+        .await
+        .unwrap();
+
+    let exists_response = client
+        .indices()
+        .exists(IndicesExistsParts::Index(&["test-index"]))
         .send()
         .await
-        .unwrap()
+        .unwrap();
+
+    assert_eq!(exists_response.status_code(), StatusCode::OK);
 }
 
-#[tokio::test(flavor = "multi_thread")]
-async fn passthrough_standard() {
-    let _compose = docker_compose("tests/test-configs/opensearch-passthrough/docker-compose.yaml");
+async fn test_index_and_search_document(client: &OpenSearch) -> String {
+    client
+        .index(IndexParts::Index("test-index"))
+        .body(json!({
+            "name": "John",
+            "age": 30
+        }))
+        .refresh(Refresh::WaitFor)
+        .send()
+        .await
+        .unwrap();
 
-    let url = Url::parse("https://localhost:9200").unwrap();
-    let credentials = Credentials::Basic("admin".into(), "admin".into());
-    let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
-        .cert_validation(CertificateValidation::None)
-        .auth(credentials)
-        .build()
+    let response = client
+        .search(SearchParts::Index(&["test-index"]))
+        .from(0)
+        .size(10)
+        .body(json!({
+            "query": {
+                "match": {
+                    "name": "John",
+                }
+            }
+        }))
+        .send()
+        .await
         .unwrap();
-    let client = OpenSearch::new(transport);
 
-    index_documents(&client).await;
+    let results = response.json::<Value>().await.unwrap();
+    assert!(results["took"].is_i64());
+    let hits = results["hits"]["hits"].as_array().unwrap();
+    assert_eq!(
+        hits.iter().map(|x| &x["_source"]).collect::<Vec<_>>(),
+        vec!(&json!({
+            "name": "John",
+            "age": 30,
+        }))
+    );
+    hits[0]["_id"].as_str().unwrap().to_owned()
+}
+
+async fn test_delete_and_search_document(client: &OpenSearch, id: String) {
+    client
```

Comment:
We should use `assert_ok_and_get_json` on all queries that we expect to succeed.